### ISSUE TITLE
NullPointerException in messaging.MessagingAnalytics.getTtl

### ISSUE THREAD

**Issue by stari4ek** [firebase/firebase-android-sdk#1820]
### [READ] Step 1: Are you in the right place?

yep

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 4.0.1
  * Firebase Component: _firebase-inappmessaging_, _firebase-inappmessaging-display_
  * Component version: 19.1.0

### [REQUIRED] Step 3: Describe the problem

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.os.Bundle.get(java.lang.String)' on a null object reference
       at com.google.firebase.messaging.MessagingAnalytics.getTtl(MessagingAnalytics.java:117)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventEncoder.encode(FirelogAnalyticsEvent.java:3)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventEncoder.encode(FirelogAnalyticsEvent.java:35)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.doEncode(JsonValueObjectEncoderContext.java:277)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:256)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.internalAdd(JsonValueObjectEncoderContext.java:317)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:70)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:32)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventWrapperEncoder.encode(FirelogAnalyticsEvent.java:4)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventWrapperEncoder.encode(FirelogAnalyticsEvent.java:6)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.doEncode(JsonValueObjectEncoderContext.java:277)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:256)
       at com.google.firebase.encoders.json.JsonDataEncoderBuilder$1.encode(JsonDataEncoderBuilder.java:121)
       at com.google.firebase.encoders.json.JsonDataEncoderBuilder$1.encode(JsonDataEncoderBuilder.java:129)
       at com.google.firebase.messaging.MessagingAnalytics.logToFirelog(MessagingAnalytics.java:106)
       at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(MessagingAnalytics.java:4)
       at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(FirebaseMessagingService.java:52)
       at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(FirebaseMessagingService.java:34)
       at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(FirebaseMessagingService.java:23)
       at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$EnhancedIntentService(EnhancedIntentService.java:43)
       at com.google.firebase.messaging.EnhancedIntentService$$Lambda$0.run(EnhancedIntentService.java:1)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at com.google.android.gms.common.util.concurrent.zza.run(zza.java:6)
       at java.lang.Thread.run(Thread.java:764)
```

#### Steps to reproduce:

I'm seeing it in crashlytics for single device only.
```
Brand: LGE
Model: Nexus 5X
Version: 8.1.0
```

**Comment by hnljp** [comment]
I get the same type of crash...
And it is happening once in a while it is always:
Brand: LGE
Model: Nexus 5X
Version: 8.1.0

And app is always in the background.

My stacktrace:
`Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.os.Bundle.get(java.lang.String)' on a null object reference
       at com.google.firebase.messaging.MessagingAnalytics.getTtl(MessagingAnalytics.java:6)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventEncoder.encode(FirelogAnalyticsEvent.java:4)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventEncoder.encode(FirelogAnalyticsEvent.java:4)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.doEncode(JsonValueObjectEncoderContext.java:7)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:317)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.internalAdd(JsonValueObjectEncoderContext.java:17)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:8)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventWrapperEncoder.encode(FirelogAnalyticsEvent.java:6)
       at com.google.firebase.messaging.FirelogAnalyticsEvent$FirelogAnalyticsEventWrapperEncoder.encode(FirelogAnalyticsEvent.java:4)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.doEncode(JsonValueObjectEncoderContext.java:7)
       at com.google.firebase.encoders.json.JsonValueObjectEncoderContext.add(JsonValueObjectEncoderContext.java:317)
       at com.google.firebase.encoders.json.JsonDataEncoderBuilder$1.encode(JsonDataEncoderBuilder.java:32)
       at com.google.firebase.encoders.json.JsonDataEncoderBuilder$1.encode(JsonDataEncoderBuilder.java:5)
       at com.google.firebase.messaging.MessagingAnalytics.logToFirelog(MessagingAnalytics.java:12)
       at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(MessagingAnalytics.java:9)
       at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(FirebaseMessagingService.java:168)
       at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(FirebaseMessagingService.java:16)
       at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(FirebaseMessagingService.java:89)
       at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$EnhancedIntentService(EnhancedIntentService.java:1)
       at com.google.firebase.messaging.EnhancedIntentService$$Lambda$0.run(EnhancedIntentService.java:6)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at com.google.android.gms.common.util.concurrent.zza.run(zza.java:7)
       at java.lang.Thread.run(Thread.java:764)`

That was with:
Firebase Analytics: 17.4.3
Firebase Messaging: 20.2.1

**Comment by gsakakihara** [comment]
@stari4ek, you listed your Firebase Component as firebase-inappmessaging, firebase-inappmessaging-display, but this looks to be a crash in firebase-messaging, is that correct?

For both @stari4ek and @hnljp, while it seems like something may be happening on that device where the extras are missing from the Intent, the crash itself looks to have been fixed with firebase-messaging version 22.0.0.

**Comment by stari4ek** [comment]
`firebase-messaging` is also used by the app.
I do not see this errors recently
